store/tikv/region_cache.go

```diff
@@ -1778,16 +1778,20 @@ func (s *Store) reResolve(c *RegionCache) (bool, error) {
 	} else {
 		metrics.RegionCacheCounterWithGetStoreOK.Inc()
 	}
-	if err != nil {
+	// when the error is "[pd] store field in rpc response not set",
+	// it means load Store from PD success but pd didn't found the store
+	// so this error should be handled by next `if` instead of here
+	if err != nil && err.Error() != "[pd] store field in rpc response not set" {
 		logutil.BgLogger().Error("loadStore from PD failed", zap.Uint64("id", s.storeID), zap.Error(err))
 		// we cannot do backoff in reResolve loop but try check other store and wait tick.
 		return false, err
 	}
-	if store == nil || store.State == metapb.StoreState_Tombstone {
```

Comment:
if the store's `State == metapb.StoreState_Tombstone`, `store` variable here will always be `nil`.